Feeder audit 2026-05-29: wallonia-issep

## Feeder audit summary for `wallonia-issep`

This sweep covers the objective repo-state parts of the feeder-release-checklist: required file presence, catalog + ghpages source entries, source-local pytest with generated producer src paths injected into PYTHONPATH, Docker E2E matrix coverage, ARM static audit, Fabric static audit, and ARM generator dry-run no-diff. It does not rerun live Azure/Fabric deploy bullets, specialist-agent reviews, or upstream API re-audits.

## Findings

- **ERROR** `ghpages-notebook-flag-missing` — 1 finding(s)
  - rts-ghpages/app.js entry does not set notebook: true even though the notebook file exists. (C:\git\rts-ghpages\app.js)
- **ERROR** `missing-docker-build-matrix-entry` — 1 finding(s)
  - tests/docker_e2e/matrix.json has no build entry for kafka (Dockerfile). (tests/docker_e2e/matrix.json)
- **ERROR** `missing-docker-flow-matrix-entry` — 1 finding(s)
  - tests/docker_e2e/matrix.json has no flow entry for kafka (Dockerfile). (tests/docker_e2e/matrix.json)
- **ERROR** `missing-file` — 2 finding(s)
  - Required file missing: feeders/wallonia-issep/xreg/wallonia-issep.xreg.json (feeders/wallonia-issep/xreg/wallonia-issep.xreg.json)
  - Required file missing: feeders/wallonia-issep/kql/wallonia-issep.kql (feeders/wallonia-issep/kql/wallonia-issep.kql)
- **ERROR** `pytest-failed` — 1 finding(s)
  - Source-local pytest failed. (feeders/wallonia-issep/tests)
- **WARNING** `arm-stub-description` — 1 finding(s)
  - azure-template-with-eventhub.json: parameter 'eventHubSku' has stub or missing metadata.description (feeders/wallonia-issep/azure-template-with-eventhub.json)
- **WARNING** `missing-image-contract` — 1 finding(s)
  - CONTAINER.md does not contain an 'Image contract' table heading. (feeders/wallonia-issep/CONTAINER.md)

## Check evidence

### `arm_audit` — rc=0, 1.27s

```text
ARM template static audit
  RepoRoot:        C:\git\real-time-sources
  Feeders scanned: 1
  Templates:       6
  Blockers:        0
  Warnings:        1

[wallonia-issep] blockers=0 warnings=3
  azure-template-with-eventhub.json
    [WARNING] stub-description: parameter 'eventHubSku' has stub or missing metadata.description
PASS (with 1 warning(s)).
Report written to C:\Users\clemensv\.copilot\session-state\cb5ea344-87f5-406e-a13a-a66811763074\files\fleet-release-audit\arm\wallonia-issep.json
```

### `arm_generator` — rc=0, 2.66s

```text
C:\git\real-time-sources\tools\.gen-out\feeders\wallonia-issep\azure-template-mqtt.json
C:\git\real-time-sources\tools\.gen-out\feeders\wallonia-issep\azure-template-with-eventgrid-mqtt.json
Description sources: total=177, container=31, readme=0, argparse=28, fallback=118
Generated 2 ARM template(s).
```

### `fabric_audit` — rc=0, 1.06s

```text
[32;1mSource        [0m[32;1m OptIn[0m[32;1m Notebook[0m[32;1m  Hook[0m[32;1m PASS[0m[32;1m WARN[0m[32;1m BLOCK[0m
[32;1m------        [0m [32;1m-----[0m [32;1m--------[0m [32;1m ----[0m [32;1m----[0m [32;1m----[0m [32;1m-----[0m
wallonia-issep  True     True False   10    0     0

Sources: 1 total, 0 blockers, 0 warnings
JSON report: C:\Users\clemensv\.copilot\session-state\cb5ea344-87f5-406e-a13a-a66811763074\files\fleet-release-audit\fabric\wallonia-issep.json
```

### `pytest` — rc=1, 3.89s

```text
============================= test session starts =============================
platform win32 -- Python 3.11.9, pytest-8.4.2, pluggy-1.6.0
rootdir: C:\git\real-time-sources\feeders\wallonia-issep
configfile: pytest.ini
plugins: anyio-4.9.0, asyncio-1.2.0, cov-6.0.0, timeout-2.4.0, requests-mock-1.12.1
asyncio: mode=Mode.STRICT, debug=False, asyncio_default_fixture_loop_scope=None, asyncio_default_test_loop_scope=function
collected 51 items

tests\test_wallonia_issep_unit.py ........F

================================== FAILURES ===================================
___________________ test_normalize_observation_full_record ____________________
tests\test_wallonia_issep_unit.py:217: in test_normalize_observation_full_record
    obs = api.normalize_observation(SAMPLE_RECORD)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
wallonia_issep\wallonia_issep.py:172: in normalize_observation
    return Observation(
C:\Users\clemensv\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\dataclasses_json\undefined.py:117: in _ignore_init
    original_init(self, **final_parameters)
E   TypeError: Observation.__init__() missing 1 required keyword-only argument: 'province'

---------- coverage: platform win32, python 3.11.9-final-0 -----------
Name                               Stmts   Miss  Cover   Missing
----------------------------------------------------------------
wallonia_issep\__init__.py             2      0   100%
wallonia_issep\__main__.py             2      2     0%   1-3
wallonia_issep\wallonia_issep.py     220    145    34%   25, 33-39, 44-50, 55-57, 63, 66-67, 73, 76-77, 115, 146-148, 152-165, 220-227, 235-256, 261-281, 289-298, 307-328, 339-379, 384-408, 417
----------------------------------------------------------------
TOTAL                                224    147    34%
Coverage HTML written to dir htmlcov

=========================== short test summary info ===========================
FAILED tests/test_wallonia_issep_unit.py::test_normalize_observation_full_record
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!
========================= 1 failed, 8 passed in 0.91s =========================
```

## Next step

Fix the findings above, then rerun the sequential feeder-release-checklist audit for this feeder and close this issue with the new evidence.

--- a/feeders/wallonia-issep/tests/test_wallonia_issep_unit.py
+++ b/feeders/wallonia-issep/tests/test_wallonia_issep_unit.py
@@ -216,6 +216,7 @@ def test_normalize_observation_full_record():
     api = WalloniaISsePAPI()
     obs = api.normalize_observation(SAMPLE_RECORD)
     assert obs.configuration_id == "10412"
+    assert obs.province == "unknown"
     assert obs.moment == "2026-04-08T20:21:04+02:00"
     assert obs.co == 16
     assert obs.no == 7
@@ -262,6 +263,7 @@ def test_normalize_observation_null_fields():
     }
     obs = api.normalize_observation(minimal_record)
     assert obs.configuration_id == "99999"
+    assert obs.province == "unknown"
     assert obs.moment == "2026-04-08T10:00:00+02:00"
     assert obs.co is None
     assert obs.no is None
@@ -291,6 +293,13 @@ def test_normalize_observation_config_id_as_string():
 
 
 @pytest.mark.unit
+def test_normalize_observation_preserves_province_when_present():
+    api = WalloniaISsePAPI()
+    obs = api.normalize_observation({**SAMPLE_RECORD, "province": "hainaut"})
+    assert obs.province == "hainaut"
+
+
+@pytest.mark.unit
 def test_normalize_observation_zero_humidity():
     """Zero humidity is valid, not None."""
     api = WalloniaISsePAPI()
@@ -310,6 +319,7 @@ def test_observation_serialization_roundtrip():
     obs = api.normalize_observation(SAMPLE_RECORD)
     serialized = obs.to_serializer_dict()
     assert serialized["configuration_id"] == "10412"
+    assert serialized["province"] == "unknown"
     assert serialized["moment"] == "2026-04-08T20:21:04+02:00"
     assert serialized["no2"] == -1
     assert serialized["pm25"] == 1.957
@@ -329,6 +339,7 @@ def test_observation_json_roundtrip():
     from wallonia_issep_producer_data import Observation
     restored = Observation.from_json(json_str)  # type: ignore[attr-defined]
     assert restored.configuration_id == "10412"
+    assert restored.province == "unknown"
     assert restored.no2 == -1
     assert restored.ppbno == 2.711
 
@@ -338,9 +349,10 @@ def test_sensor_configuration_serialization():
     from wallonia_issep_producer_data import SensorConfiguration
     config = SensorConfiguration(configuration_id="10412")
     serialized = config.to_serializer_dict()
-    assert serialized == {"configuration_id": "10412"}
+    assert serialized == {"configuration_id": "10412", "province": "unknown"}
     restored = SensorConfiguration.from_serializer_dict(serialized)
     assert restored.configuration_id == "10412"
+    assert restored.province == "unknown"
 
 
 @pytest.mark.unit
@@ -361,6 +373,7 @@ def test_observation_byte_array_roundtrip():
     restored = Observation.from_data(byte_arr, "application/json")
     assert restored is not None
     assert restored.configuration_id == "10412"
+    assert restored.province == "unknown"
     assert restored.no2 == -1
 
 
@@ -372,6 +385,7 @@ def test_observation_null_fields_in_json():
     obs = api.normalize_observation(minimal)
     json_str = obs.to_json()  # type: ignore[attr-defined]
     parsed = json.loads(json_str)
+    assert parsed["province"] == "unknown"
     assert parsed["co"] is None
     assert parsed["pm10"] is None
     assert parsed["vbat"] is None
@@ -393,6 +407,17 @@ def test_extract_configurations():
     assert len(configs) == 2
     ids = {c.configuration_id for c in configs}
     assert ids == {"10412", "10296"}
+    assert {c.province for c in configs} == {"unknown"}
+
+
+@pytest.mark.unit
+def test_extract_configurations_preserves_province():
+    records = [
+        {"id_configuration": 10412, "moment": "2026-04-08T20:21:04+02:00", "province": "hainaut"},
+    ]
+    configs = WalloniaISsePAPI.extract_configurations(records)
+    assert configs[0].configuration_id == "10412"
+    assert configs[0].province == "hainaut"
 
 
 @pytest.mark.unit

--- a/feeders/wallonia-issep/wallonia_issep/wallonia_issep.py
+++ b/feeders/wallonia-issep/wallonia_issep/wallonia_issep.py
@@ -85,6 +85,12 @@ def _safe_float(value: Any) -> Optional[float]:
         return None
 
 
+def _province_from_record(record: Dict[str, Any]) -> str:
+    """Return the Walloon province slug when present, otherwise the contract sentinel."""
+    province = record.get("province")
+    return str(province).strip() if province else "unknown"
+
+
 class WalloniaISsePAPI:
     """Poll the Wallonia ISSeP Opendatasoft API and emit CloudEvents."""
 
@@ -177,9 +183,11 @@ class WalloniaISsePAPI:
         """Normalize one upstream Opendatasoft record into an Observation."""
         config_id = str(record.get("id_configuration", ""))
         moment = str(record.get("moment", ""))
+        province = _province_from_record(record)
 
         return Observation(
             configuration_id=config_id,
+            province=province,
             moment=moment,
             co=_safe_int(record.get("co")),
             no=_safe_int(record.get("no")),
@@ -232,7 +240,9 @@ class WalloniaISsePAPI:
             config_id = str(record.get("id_configuration", ""))
             if config_id and config_id not in seen:
                 seen.add(config_id)
-                configs.append(SensorConfiguration(configuration_id=config_id))
+                configs.append(
+                    SensorConfiguration(configuration_id=config_id, province=_province_from_record(record))
+                )
         return configs
 
     @staticmethod

